### PR TITLE
Skillonomy 58

### PR DESCRIPTION
**Pull request by RuslanSkiraRaccoongang**
**DO NOT MERGE I WILL SQUASH COMMITS**
**Description:** Add user stats fetched from customer's platform

[**Youtrack SKILLONOMY-58:**](https://youtrack.raccoongang.com/issue/SKILLONOMY-58)

**Reviewers:**
- [ ] @oleksandra-martyntseva
- [ ] @OPersian  

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Documentation in source code updated
- [ ] All related Confluence documentation is updated (including deployment documentation)
- [ ] Commits are squashed

### CODE REVIEW

File: common/djangoapps/edeos/api_calls.py [hunk 1]
```diff
@@ -140,9 +145,13 @@ def post(self, url, payload, headers=None, can_retry=True):
         if headers is not None:
             headers_.update(headers)
         resp = requests.post(url, json=payload, headers=headers_)
-        log.info("Edeos response: status {}, content {}".format(resp.status_code, resp.content))
+        log.info("Edeos response: status {}, content {}".format(resp.status_code, resp.content)) #TODO this response will be in the termilall and it is not convenient maybe not in the teminal?
```
Comment (OPersian) [inline common/djangoapps/edeos/api_calls.py]: Please explain why this log isn't convenient; not sure I understand...
Or maybe remove this TODO :)

File: common/djangoapps/edeos/api_calls.py [hunk 2]
```diff
@@ -140,9 +145,13 @@ def post(self, url, payload, headers=None, can_retry=True):
...
             headers_.update(headers)
         resp = requests.post(url, json=payload, headers=headers_)
-        log.info("Edeos response: status {}, content {}".format(resp.status_code, resp.content))
+        log.info("Edeos response: status {}, content {}".format(resp.status_code, resp.content)) #TODO this response will be in the termilall and it is not convenient maybe not in the teminal? 
+        
         if resp.status_code in (httplib.OK, httplib.CREATED):
-            return resp.json()
+            try:
+                return resp.json()
+            except JSONDecodeError:
```
Comment (OPersian) [inline common/djangoapps/edeos/api_calls.py]: Just a note, for future references: couldn't avoid this try-except here, as profile_statistics is very peculiar (for now, it's a single endpoint that doesn't return JSON )

File: common/djangoapps/edeos/api_calls.py [hunk 3]
```diff
@@ -182,12 +191,14 @@ def call_api(self, endpoint_url, payload):
                 url="{}{}".format(self.base_url, endpoint_url),
                 payload=payload)
             return response
+        # TODO lines was commented 191 and 196-202
```
Comment (OPersian) [inline common/djangoapps/edeos/api_calls.py]: I guess this TODO should be removed (looks like it was used for debugging only)

File: common/djangoapps/edeos/api_calls.py [hunk 4]
```diff
@@ -246,4 +257,12 @@ def referrals_store(self, payload):
         return self.call_api("referrals/store", payload)
 
     def statistics(self, payload):
-        return self.call_api("statistics", payload)
\ No newline at end of file
+        return self.call_api("statistics", payload)
+    
+    def profile_statistics(self, payload):
+        return self.call_api("profile_statistics", payload) 
+
```
Comment (OPersian) [inline common/djangoapps/edeos/api_calls.py]: Please consider removing unnecessary lines

File: lms/static/js/student_account/views/account_settings_factory.js [hunk 5]
```diff
@@ -216,7 +232,44 @@
                 );
             }
 
+
+
             aboutSectionsData.push(socialFields);
+            // Add the usersID and mobytizetoken fields
```
Comment (OPersian) [inline lms/static/js/student_account/views/account_settings_factory.js]: usersID - why plural?

File: lms/static/js/student_account/views/account_settings_factory.js [hunk 6]
```diff
@@ -216,7 +232,44 @@
...
+                        helpMessage: gettext(
+                            'mobytize_token'  // eslint-disable-line max-len
+                        ),
+                        persistChanges: true
+                    })
+                }]
+            };
+            aboutSectionsData.push(mobytizeFields);
+
+
```
Comment (OPersian) [inline lms/static/js/student_account/views/account_settings_factory.js]: Please remove unnecessary lines

File: lms/templates/student_account/account_settings.underscore [hunk 7]
```diff
@@ -15,5 +16,8 @@
             <div id="<%- tab.name %>-tabpanel" class="account-settings-tabpanels <% if (!tab.class) { %> hidden <% } %>" aria-label="<%- tab.label %>" role="tabpanel"></div>
         <% }); %>
     </div>
+
+    <h1>Where I could FIND?????</h1>
```
Comment (OPersian) [inline lms/templates/student_account/account_settings.underscore]: I believe this should be removed :)

File: lms/templates/student_profile/learner_performance.html [hunk 8]
```diff
@@ -0,0 +1,5 @@
+%if data['profile_statistics']:
+    ${data['profile_statistics'].content}
+%else :
+     <h1>No data ${data['profile_statistics']}</h1>
+% endif
```
Comment (OPersian) [inline lms/templates/student_profile/learner_performance.html]: Please add a new line at the end of the file (here and in other cases, whenever missing)

Comment (OPersian) [reply lms/templates/student_profile/learner_performance.html]: ...for the sake of this: https://unix.stackexchange.com/a/18789

File: lms/templates/student_profile/learner_statistic.html [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<%page expression_filter="h" />
+<%inherit file="/main.html" />
+<%!
+import json
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext as _
+from openedx.core.djangolib.js_utils import dump_js_escaped_json
+%>
+<iframe src="https://vimeo.com/stock/collection/5855140"></iframe>
```
Comment (OPersian) [inline lms/templates/student_profile/learner_statistic.html]: Was it intended for debugging? Please remove if so

File: common/djangoapps/edeos/api_calls.py (hunk 4 above)
Comment (OPersian) [reply common/djangoapps/edeos/api_calls.py]: (here and in other cases whenever applicable)

File: lms/urls.py [hunk 10]
```diff
@@ -631,6 +646,23 @@
...
+        # Student performance
+    url(
+        r'^/performance/'.format(
+            username_pattern=settings.USERNAME_PATTERN,
+        ),
+        'student_profile.views.learner_performance',
+        name='learner_performance',
+    ),
+    # Student profile_statistics
+    # TODO remover the user pattern. Security. Get user from session. As everywhere else.
```
Comment (OPersian) [inline lms/urls.py]: I guess this TODO should be removed as a user from the session is current, or viewing, user, while {username} is the user being viewed, right? (as we discussed)

File: lms/static/js/student_account/views/account_settings_factory.js [hunk 11]
```diff
@@ -186,7 +186,23 @@
...
                             })
-                        }
+                        }, 
+                        {
+                            view: new AccountSettingsFieldViews.TextFieldView({
+                            model: userAccountModel,
+                            title: gettext('Full Name'),
+                            valueAttribute: 'name',
+                            helpMessage: gettext(
+                                'The name that is used for ID verification and appears on your certificates. Other learners never see your full name. Make sure to enter your name exactly as it appears on your government-issued photo ID, including any non-Roman characters.'  // eslint-disable-line max-len
```
Comment (OPersian) [inline lms/static/js/student_account/views/account_settings_factory.js]: Not sure where this diff comes from, looks unrelated to the current task. Please ignore this comment if I'm wrong

File: openedx/core/djangoapps/user_api/accounts/api.py [hunk 12]
```diff
@@ -241,6 +241,7 @@ def update_account_settings(requesting_user, update, username=None):
                 u"Error thrown from do_email_change_request: '{}'".format(err.message),
                 user_message=err.message
             )
+
```
Comment (oleksandra-martyntseva) [inline openedx/core/djangoapps/user_api/accounts/api.py]: Why do we need this empty line?

File: common/djangoapps/edeos/migrations/0002_auto_20190723_0226.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+
+    dependencies = [
+        ('edeos', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='usersociallink',
+            name='platform',
+            field=models.CharField(max_length=50, choices=[(b'facebook', b'Facebook')]),
```
Comment (OPersian) [inline common/djangoapps/edeos/migrations/0002_auto_20190723_0226.py]: Looks like we missed this migration before :( 
here: https://github.com/raccoongang/edx-platform/commit/fb96b6537727fabf0ca56671ecb514b6d5811c97

File: common/djangoapps/student/models.py [hunk 14]
```diff
@@ -294,6 +294,9 @@ class Meta(object):
     allow_certificate = models.BooleanField(default=1)
     bio = models.CharField(blank=True, null=True, max_length=3000, db_index=False)
     profile_image_uploaded_at = models.DateTimeField(null=True, blank=True)
+    mobytize_id = models.CharField(max_length=255, blank=True)
```
Comment (OPersian) [inline common/djangoapps/student/models.py]: I believe we need another migration for these new fields (mobytize_id, mobytize_token)

File: lms/djangoapps/student_profile/views.py [hunk 15]
```diff
@@ -120,3 +124,76 @@ def learner_profile_context(request, profile_username, user_is_staff):
...
+    """
+        view for maiking request on mobytize platform
+    """
+    if student_id:
+        student = User.objects.get(id=int(student_id))
+    else:
+        student = request.user
+
+    date_from = CourseEnrollment.objects.filter(user=student.id).filter(is_active=True).order_by('created').first().created
+    date_to = CourseEnrollment.objects.filter(user=student.id).filter(is_active=True).order_by('created').last().created
```
Comment (oleksandra-martyntseva) [inline lms/djangoapps/student_profile/views.py]: The date picking logic here is wrong.

File: common/djangoapps/student/models.py (hunk 14 above)
Comment (OPersian) [reply common/djangoapps/student/models.py]: this one: https://github.com/raccoongang/edx-platform/blob/d4f397584f679c111c0af141f7ee602a4a418597/common/djangoapps/student/migrations/0011_auto_20190726_0445.py

File: common/djangoapps/edeos/api_calls.py (hunk 1 above)
Comment (OPersian) [reply common/djangoapps/edeos/api_calls.py]: resolved

File: lms/static/js/student_account/views/account_settings_factory.js (hunk 6 above)
Comment (OPersian) [reply lms/static/js/student_account/views/account_settings_factory.js]: resolved

File: lms/templates/student_account/account_settings.underscore (hunk 7 above)
Comment (OPersian) [reply lms/templates/student_account/account_settings.underscore]: resolved

File: lms/templates/student_profile/learner_performance.html (hunk 8 above)
Comment (OPersian) [reply lms/templates/student_profile/learner_performance.html]: resolved

File: lms/static/js/student_account/views/account_settings_factory.js (hunk 11 above)
Comment (OPersian) [reply lms/static/js/student_account/views/account_settings_factory.js]: resolved

File: lms/djangoapps/student_profile/views.py [hunk 16]
```diff
@@ -120,3 +124,74 @@ def learner_profile_context(request, profile_username, user_is_staff):
...
+        "key": getattr(settings, 'EDEOS_API_KEY'),
+        "secret": getattr(settings, 'EDEOS_API_SECRET'),
+        "base_url": 'http://195.160.222.156/api/point/v1/'
+            }
+
+    response = send_edeos_api_request(**edeos_post_data)
+
+    try:
+        return render_to_response(
+            'student_profile/learner_performance.html',  # TODO add logic to the template - if no profile_statistics was fetched, display error_message
```
[the remark this answers is not in the document]
Comment (OPersian) [reply lms/djangoapps/student_profile/views.py]: This TODO was part of requirements as far as I know.
Looks like it's not resolved yet...

Shouldn't it be resolved with this PR?
@RuslanSkiraRaccoongang

[the remark this answers is not in the document]
Comment (OPersian) [reply lms/djangoapps/student_profile/views.py]: The req:

> 6. В ответе на этот запрос гейт будет передавать статус ответа и html код страницы, который нужно будет вывести на странице статистики. Если статус запроса “ошибка” - то нужно будет вывести сообщение об ошибке для пользователя. Если статус ответа "ок" - тогда необходимо вывести полученные данные.


ref https://raccoongang.slack.com/archives/CDQ7E16RX/p1562254428027500?thread_ts=1561986550.020800&cid=CDQ7E16RX

File: common/djangoapps/student/models.py (hunk 14 above)
Comment (RuslanSkiraRaccoongang) [reply common/djangoapps/student/models.py]: common/djangoapps/student/migrations/0011_auto_20190726_1017.py added to git

File: lms/djangoapps/student_profile/views.py [hunk 17]
```diff
@@ -120,3 +124,74 @@ def learner_profile_context(request, profile_username, user_is_staff):
         context['data']['badges_api_url'] = reverse("badges_api:user_assertions", kwargs={'username': profile_username})
 
     return context
+
+def learner_profile_statistics_context(profile_username, profile_statistics):
+    """Context for the learner profile page.
+
+    Args:
+        logged_in_user (object): Logged In user.
```
[the remark this answers is not in the document]
Comment (RuslanSkiraRaccoongang) [reply lms/djangoapps/student_profile/views.py]: fixed

File: lms/djangoapps/student_profile/views.py (hunk 15 above)
Comment (RuslanSkiraRaccoongang) [reply lms/djangoapps/student_profile/views.py]: I changed logic of datate_to

File: common/djangoapps/edeos/migrations/0002_auto_20190723_0226.py (hunk 13 above)
Comment (RuslanSkiraRaccoongang) [reply common/djangoapps/edeos/migrations/0002_auto_20190723_0226.py]: I add the migration

File: lms/djangoapps/student_profile/views.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (oleksandra-martyntseva) [reply lms/djangoapps/student_profile/views.py]: currently the api always returns data =( if something went wrong, it just returns test data for test user =( unfortunately this is out of our hands

File: common/djangoapps/edeos/api_calls.py [hunk 18]
```diff
@@ -246,4 +252,8 @@ def referrals_store(self, payload):
         return self.call_api("referrals/store", payload)
 
     def statistics(self, payload):
-        return self.call_api("statistics", payload)
\ No newline at end of file
+        return self.call_api("statistics", payload)
+    
+    def profile_statistics(self, payload):
+        return self.call_api("profile_statistics", payload)
+
```
Comment (oleksandra-martyntseva) [inline common/djangoapps/edeos/api_calls.py]: please fix the no newline at the end of file

File: common/djangoapps/student/migrations/0011_auto_20190726_1017.py [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+    operations = [
+        migrations.AddField(
+            model_name='userprofile',
+            name='mobytize_id',
+            field=models.CharField(max_length=255, blank=True),
+        ),
+        migrations.AddField(
+            model_name='userprofile',
+            name='mobytize_token',
+            field=models.CharField(max_length=255, blank=True),
```
Comment (oleksandra-martyntseva) [inline common/djangoapps/student/migrations/0011_auto_20190726_1017.py]: please make it max_length=64

File: common/djangoapps/student/migrations/0011_auto_20190726_1017.py [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+
+    dependencies = [
+        ('student', '0010_auto_20170207_0458'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='userprofile',
+            name='mobytize_id',
+            field=models.CharField(max_length=255, blank=True),
```
Comment (oleksandra-martyntseva) [inline common/djangoapps/student/migrations/0011_auto_20190726_1017.py]: please make it max_length=32

File: conf/locale/ru/LC_MESSAGES/django.po [hunk 21]
```diff
@@ -24723,4 +24727,4 @@ msgid ""
 msgstr ""
 "Если Вы не создавали учетную запись, дальнейших действий не требуется; "
 "Вы больше не будете получать от нас сообщения. Если Вам потребуется помощь, пожалуйста, не отвечайте на это сообщение. "
-"Обратитесь к разделу \"Помощь\" на веб-сайте {platform_name}."
+"Обратитесь к разделу \"Помощь\" на веб-сайте {platform_name}."
```
Comment (oleksandra-martyntseva) [inline conf/locale/ru/LC_MESSAGES/django.po]: please fix the no newline at the end of file

File: lms/djangoapps/courseware/views/views.py [hunk 22]
```diff
@@ -1787,3 +1787,4 @@ def check_access_to_course(request, course):
     # Redirect if the user must answer a survey before entering the course.
     if must_answer_survey(course, request.user):
         raise CourseAccessRedirect(reverse('course_survey', args=[unicode(course.id)]))
+
```
Comment (oleksandra-martyntseva) [inline lms/djangoapps/courseware/views/views.py]: please fix the no newline at the end of file

File: lms/djangoapps/student_profile/views.py [hunk 23]
```diff
@@ -120,3 +124,74 @@ def learner_profile_context(request, profile_username, user_is_staff):
...
+        dict
+    """
+    context = {
+        'data': {
+            'error_message': '',
+            'profile_statistics': profile_statistics,
+            'username': profile_username,
+            },
+    }
+
```
Comment (oleksandra-martyntseva) [inline lms/djangoapps/student_profile/views.py]: Too many blank lines here

File: lms/djangoapps/student_profile/views.py [hunk 24]
```diff
@@ -120,3 +124,74 @@ def learner_profile_context(request, profile_username, user_is_staff):
...
+    """
+    if student_id:
+        student = User.objects.get(id=int(student_id))
+    else:
+        student = request.user
+
+    date_from = CourseEnrollment.objects.filter(user=student.id).filter(is_active=True).order_by('created').first().created
+
+    from datetime import datetime
+    if date_from:
```
Comment (oleksandra-martyntseva) [inline lms/djangoapps/student_profile/views.py]: Why is this import here? Can there be no date_from? If so, why are we using it in line 165 without any checks?

File: lms/djangoapps/student_profile/views.py [hunk 25]
```diff
@@ -120,3 +124,74 @@ def learner_profile_context(request, profile_username, user_is_staff):
...
+        date_to = datetime.today()
+
+    date_to = date_to.strftime("%Y-%m-%d")
```
Comment (oleksandra-martyntseva) [inline lms/djangoapps/student_profile/views.py]: Based on the code above, the date_to can be None here.

File: lms/djangoapps/student_profile/views.py [hunk 26]
```diff
@@ -120,3 +124,74 @@ def learner_profile_context(request, profile_username, user_is_staff):
...
+    date_from = date_from.strftime("%Y-%m-%d")
+
+    edeos_post_data = {
+        "payload": {
+            "date_from":date_from,
+	        "date_to":date_to,
```
Comment (oleksandra-martyntseva) [inline lms/djangoapps/student_profile/views.py]: formatting

File: lms/djangoapps/student_profile/views.py [hunk 27]
```diff
@@ -120,3 +124,74 @@ def learner_profile_context(request, profile_username, user_is_staff):
...
+            "type": 0 if request.user.is_staff else 1,
+            "mobytize_token":student.profile.mobytize_token,
+            "users":student.profile.mobytize_id
+        },
+        "api_endpoint": "profile_statistics",
+        "key": getattr(settings, 'EDEOS_API_KEY'),
+        "secret": getattr(settings, 'EDEOS_API_SECRET'),
+        "base_url": 'http://195.160.222.156/api/point/v1/'
+            }
```
Comment (oleksandra-martyntseva) [inline lms/djangoapps/student_profile/views.py]: brace indentation

File: lms/djangoapps/student_profile/views.py [hunk 28]
```diff
@@ -120,3 +124,74 @@ def learner_profile_context(request, profile_username, user_is_staff):
...
+            "date_from":date_from,
+	        "date_to":date_to,
+            "type": 0 if request.user.is_staff else 1,
+            "mobytize_token":student.profile.mobytize_token,
+            "users":student.profile.mobytize_id
+        },
+        "api_endpoint": "profile_statistics",
+        "key": getattr(settings, 'EDEOS_API_KEY'),
+        "secret": getattr(settings, 'EDEOS_API_SECRET'),
+        "base_url": 'http://195.160.222.156/api/point/v1/'
```
Comment (oleksandra-martyntseva) [inline lms/djangoapps/student_profile/views.py]: maybe this should also be in the settings?

File: lms/djangoapps/student_profile/views.py [hunk 29]
```diff
@@ -120,3 +124,74 @@ def learner_profile_context(request, profile_username, user_is_staff):
...
+
+    date_to = date_to.strftime("%Y-%m-%d")
+    date_from = date_from.strftime("%Y-%m-%d")
+
+    edeos_post_data = {
+        "payload": {
+            "date_from":date_from,
+	        "date_to":date_to,
+            "type": 0 if request.user.is_staff else 1,
+            "mobytize_token":student.profile.mobytize_token,
```
Comment (oleksandra-martyntseva) [inline lms/djangoapps/student_profile/views.py]: we should make sure the student actually has the mobytize token and id before sending the request

File: common/djangoapps/edeos/migrations/0002_auto_20190723_0226.py (hunk 13 above)
Comment (RuslanSkiraRaccoongang) [reply common/djangoapps/edeos/migrations/0002_auto_20190723_0226.py]: ok I will add this Migration in this tasks.

File: lms/templates/student_account/account_settings.underscore [hunk 30]
```diff
@@ -2,6 +2,7 @@
 <div class="account-settings-container">
     <div class="wrapper-header">
         <h2 class="header-title"><%- gettext("Account Settings") %></h2>
+
```
Comment (oleksandra-martyntseva) [inline lms/templates/student_account/account_settings.underscore]: If the whitespace change is the only change in a file - you should not do it

File: lms/templates/student_profile/learner_profile.html [hunk 31]
```diff
@@ -22,6 +22,9 @@
     </div>
     <%include file="/edeos/update_wallet.html"/>
 </main>
+% if data['active_course']: 
+    <button type="button" style="margin-left: 400px;"><a href="${data['account_settings_data']['username']}/profile_statistics">${_("Profile Statistics")}</a></button>
```
[the remark this answers is not in the document]
Comment (oleksandra-martyntseva) [reply lms/templates/student_profile/learner_profile.html]: is there a security check? What if I as a student would just paste this link into my browser? Would I see someone else's data? I should not see it, I should see 403

File: lms/djangoapps/student_profile/views.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (RuslanSkiraRaccoongang) [reply lms/djangoapps/student_profile/views.py]: ok

File: lms/templates/student_profile/learner_profile.html (hunk 31 above)
[the remark this answers is not in the document]
Comment (RuslanSkiraRaccoongang) [reply lms/templates/student_profile/learner_profile.html]: image
Other student couldn't see the data from the other student link

File: lms/urls.py [hunk 32]
```diff
@@ -631,6 +646,23 @@
...
+        r'^/performance/'.format(
+            username_pattern=settings.USERNAME_PATTERN,
+        ),
+        'student_profile.views.learner_performance',
+        name='learner_performance',
+    ),
+    # Student profile_statistics
+    # TODO remover the user pattern. Security. Get user from session. As everywhere else. 
+    url(
+        r'^u/{username_pattern}/profile_statistics'.format(
```
[the remark this answers is not in the document]
Comment (RuslanSkiraRaccoongang) [reply lms/urls.py]: image
Student couldn't use link from other student

File: common/djangoapps/edeos/migrations/0002_auto_20190723_0226.py (hunk 13 above)
Comment (RuslanSkiraRaccoongang) [reply common/djangoapps/edeos/migrations/0002_auto_20190723_0226.py]: ok

File: common/djangoapps/edeos/api_calls.py (hunk 18 above)
Comment (RuslanSkiraRaccoongang) [reply common/djangoapps/edeos/api_calls.py]: But it should be there? or shouldn't?

File: conf/locale/ru/LC_MESSAGES/django.po (hunk 21 above)
Comment (RuslanSkiraRaccoongang) [reply conf/locale/ru/LC_MESSAGES/django.po]: I added the line

File: lms/djangoapps/courseware/views/views.py (hunk 22 above)
Comment (RuslanSkiraRaccoongang) [reply lms/djangoapps/courseware/views/views.py]: I added the line at the end of the file

**Comment by RuslanSkiraRaccoongang** [comment]
https://github.com/raccoongang/edx-platform/pull/1403#discussion_r308230907 Student will not see 403. He will see loging page and iframe with mistake
image

File: lms/templates/student_account/account_settings.underscore (hunk 30 above)
Comment (RuslanSkiraRaccoongang) [reply lms/templates/student_account/account_settings.underscore]: Fixed

File: lms/djangoapps/student_profile/views.py (hunk 29 above)
Comment (RuslanSkiraRaccoongang) [reply lms/djangoapps/student_profile/views.py]: I added check in template student_profile/learner_profile.html 
If student don't have mobytize token or mobytize id he wouldn't see the button

File: lms/templates/student_profile/learner_profile.html (hunk 31 above)
[the remark this answers is not in the document]
Comment (oleksandra-martyntseva) [reply lms/templates/student_profile/learner_profile.html]: created a task to do this later https://youtrack.raccoongang.com/issue/SKILLONOMY-119

File: common/djangoapps/edeos/api_calls.py (hunk 18 above)
Comment (oleksandra-martyntseva) [reply common/djangoapps/edeos/api_calls.py]: there should. Check out the next file - there is no such error there

File: common/djangoapps/student/migrations/0011_auto_20190726_1017.py (hunk 20 above)
Comment (RuslanSkiraRaccoongang) [reply common/djangoapps/student/migrations/0011_auto_20190726_1017.py]: fixed

File: common/djangoapps/student/migrations/0011_auto_20190726_1017.py (hunk 19 above)
Comment (RuslanSkiraRaccoongang) [reply common/djangoapps/student/migrations/0011_auto_20190726_1017.py]: fixed

File: lms/djangoapps/student_profile/views.py (hunk 23 above)
Comment (RuslanSkiraRaccoongang) [reply lms/djangoapps/student_profile/views.py]: fixed
